clouddriver-kubernetes/src/main/java/com/netflix/spinnaker/clouddriver/kubernetes/security/KubernetesCredentials.java

```diff
@@ -313,14 +318,22 @@ public KubernetesKindProperties getKindProperties(@Nonnull KubernetesKind kind)
       return ImmutableMap.of();
     }
     try {
-      return list(KubernetesKind.CUSTOM_RESOURCE_DEFINITION, "").stream()
-          .map(
-              manifest ->
-                  KubernetesCacheDataConverter.getResource(
-                      manifest, V1beta1CustomResourceDefinition.class))
-          .map(KubernetesKindProperties::fromCustomResourceDefinition)
-          .collect(
-              toImmutableMap(KubernetesKindProperties::getKubernetesKind, Function.identity()));
+      ImmutableMap<KubernetesKind, KubernetesKindProperties> crds =
+          list(KubernetesKind.CUSTOM_RESOURCE_DEFINITION, "").stream()
+              .map(
+                  manifest ->
+                      KubernetesCacheDataConverter.getResource(
+                          manifest, V1beta1CustomResourceDefinition.class))
+              .map(KubernetesKindProperties::fromCustomResourceDefinition)
+              .collect(
+                  toImmutableMap(KubernetesKindProperties::getKubernetesKind, Function.identity()));
+
+      for (KubernetesKind kind : crds.keySet()) {
```

Comment:
Nit: you can use streams to update the list in `globalResourcePropertyRegistry`:
```
List<KubernetesHandler> crdHandlers = crds.keySet().stream()
  .map(KubernetesCustomResourceHandler::new)
  .collect(toImmutableList());
this.globalResourcePropertyRegistry.updateCrdProperties(crdHandlers);
```